### PR TITLE
fix #2017 issue with move and default value to undefined

### LINKED ISSUE #2017: Nested Field Array Not Moving Errors
**Describe the bug**
Nested Field Array Not Moving Errors
https://spectrum.chat/react-hook-form/help/nested-field-array-not-moving-errors~da833f41-0e19-4b32-adb8-f22feb97800a

https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-e616x?file=/src/nestedFieldArray.js.

**To reproduce the issue I am seeing:**
- Click 'Append Nested' twice
- Put a value into 'Sub Nested Array'[0] input
- Click in and out of 'Sub Nested Array'[1] input to trigger a blur error (input will have 'ERROR' in red under it)
- Click the 'Move Nested' button (hard coded to move 'Nested Array'[0] to 'Nested Array'[1])

**What happens:**
- 'Sub Nested Array' inputs swap
- 'Sub Nested Array'[1] error stays at 'Sub Nested Array'[1]

**What is expected:**
'Sub Nested Array'[1] error moves to 'Sub Nested Array'[0] as the value does
Any ideas on what I am doing wrong?

### PR DESCRIPTION
**Pull request by bluebill1049**
- [x] fix move with empty slot
- [x] change array empty slot from `null` to `undefined`

fix #2017

https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-vst22?file=/src/src/useFieldArray.ts

### CODE REVIEW

**Comment by bluebill1049** [comment]
Thanks you @keiya01 🙏

File: src/utils/insert.test.ts [hunk 1]
```diff
@@ -54,11 +54,16 @@ describe('insert', () => {
   });
 
   it('should insert null as value when value to be inserted is falsy', () => {
```
Comment (keiya01) [inline src/utils/insert.test.ts:56]: Should you change this word ?

```suggestion
  it('should insert undefined as value when value to be inserted is falsy', () => {
```

Comment (bluebill1049) [reply src/utils/insert.test.ts]: yep!

**Review by keiya01 — APPROVED** [review]
LGTM 🙆‍♂️

**Pull request merged**